script.js

```diff
@@ -1,12 +1,327 @@
+    containerShow.className = "container-show";
+    divShow.appendChild(containerShow);
+
+    let divImage = document.createElement("div");
+    containerShow.appendChild(divImage);
+
+    let imageShow = document.createElement("img");
+    divImage.appendChild(imageShow);
+
+    if (show.image === null) {
+      imageShow.src =
+        "https://upload.wikimedia.org/wikipedia/commons/0/0a/No-image-available.png";
+    } else {
+      imageShow.src = show.image.medium;
+    }
+
+    let summaryDiv = document.createElement("div");
+    summaryDiv.className = "p-div-show";
+    containerShow.appendChild(summaryDiv);
+
+    let showSummary = document.createElement("p");
+    showSummary.className = "show-summary";
+    if (show.summary === null) {
+      showSummary.innerText = "No Summary";
+    } else {
+      showSummary.innerText = show.summary
+        .replace(/<p>/g, "")
+        .replace(/<\/p>/g, "")
+        .replace(/<b>/g, "")
+        .replace(/<\/b>/g, "")
+        .replace(/<i>/g, "")
+        .replace(/<\/i>/g, "");
+    }
+    summaryDiv.appendChild(showSummary);
+
+    let detailsDiv = document.createElement("div");
+    detailsDiv.className = "details-div";
+    containerShow.appendChild(detailsDiv);
+    detailsDiv.innerHTML = `
+      <p class="rating">Rated: ${show.rating.average}</p>
+      <p>Genres: ${show.genres.join(" /")}</p>
+      <p>Status: ${show.status}</p>
+      <p>Runtime: ${show.runtime}</p>
+    `;
+  });
+}
+
+function showEpisodesWhenClickOnShow() {
+  //search input
+  let labelInput = document.createElement("label");
+  divSearch.appendChild(labelInput);
+  labelInput.innerText = "Search";
+  labelInput.htmlFor = "input-search";
+
+  let input = document.createElement("input");
+  divSearch.appendChild(input);
+  input.id = "input-search";
+  input.setAttribute("type", "text");
+  input.addEventListener("keyup", searchInEpisodes);
+
+  //displaying the number of episodes
+  let numOfEpisodes = document.createElement("p");
+  numOfEpisodes.className = "num-episodes";
+  numOfEpisodes.id = "number-episodes";
+  divSearch.appendChild(numOfEpisodes);
+
+  makeSelect(allShows);
+
+  document.getElementById("container-div").innerHTML = "";
+
+  let divSearchAll = document.getElementById("search-container");
+  divSearchAll.style.display = "flex";
+
+  let newShowUrl = url.replace("show-id", `${this.id}`);
+  fetchEpisodes(newShowUrl);
+
+  document.getElementById(`${this.id}`).selected = true;
+  document.getElementById("div-filter-show").innerHTML = "";
+}
+
+function filterInShows() {
+  document.getElementById("container-div").innerHTML = "";
+  document.getElementById("selectShow").innerHTML = "";
+  valueFilter = inputFilterShows.value.toLowerCase();
+  const filterShows = allShows.filter(
+    (show) =>
+      show.summary.toLowerCase().includes(valueFilter) ||
+      show.name.toLowerCase().includes(valueFilter) ||
+      show.genres.includes(valueFilter)
+  );
```

Comment:
this filter is a particular detailed search according to the requirement, and it is really nice